components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/model/ServerURL.java

```diff
@@ -0,0 +1,411 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.core.model;
+
+import org.apache.axis2.engine.AxisConfiguration;
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.base.ServerConfiguration;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.core.URLResolverException;
+import org.wso2.carbon.identity.core.internal.IdentityCoreServiceComponent;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.utils.CarbonUtils;
+import org.wso2.carbon.utils.NetworkUtils;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.net.SocketException;
+import java.util.ArrayList;
+
+import static org.wso2.carbon.identity.core.util.IdentityTenantUtil.isTenantQualifiedUrlsEnabled;
+
+/**
+ * Server URL representation.
+ */
+public class ServerURL {
+
+    private String protocol;
+    private String hostName;
+    private Integer port;
+    private String proxyContext;
+    private String webContext;
+    private Boolean addProxyContext;
+    private Boolean addWebContext;
+    private String urlContext;
+    private Boolean appendTenantAsQueryParam;
+    private Boolean appendTenantAsPathParam;
```

Comment:
I don't think these properties are needed. Those are just builder properties.
We need just the basic components of the URL like host, port, protocol, path, query, fragement, parameters. May be you can have an internal URL object